src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs

```diff
@@ -15,11 +15,18 @@ public static class RateLimiterOptionsExtensions
     /// </summary>
     /// <param name="options">The <see cref="RateLimiterOptions"/> to add a limiter to.</param>
     /// <param name="policyName">The name that will be associated with the limiter.</param>
-    /// <param name="tokenBucketRateLimiterOptions">The <see cref="TokenBucketRateLimiterOptions"/> to be used for the limiter.</param>
+    /// <param name="configureOptions">A callback to configure the <see cref="TokenBucketRateLimiterOptions"/> to be used for the limiter.</param>
     /// <returns>This <see cref="RateLimiterOptions"/>.</returns>
-    public static RateLimiterOptions AddTokenBucketLimiter(this RateLimiterOptions options, string policyName, TokenBucketRateLimiterOptions tokenBucketRateLimiterOptions)
+    public static RateLimiterOptions AddTokenBucketLimiter(this RateLimiterOptions options, string policyName, Action<TokenBucketRateLimiterOptions> configureOptions)
     {
+        if (configureOptions is null)
+        {
+            throw new ArgumentNullException(nameof(configureOptions));
+        }
+
         var key = new PolicyNameKey() { PolicyName = policyName };
+        var tokenBucketRateLimiterOptions = new TokenBucketRateLimiterOptions();
+        configureOptions.Invoke(tokenBucketRateLimiterOptions);
```

Comment:
nit: you could force `AutoReplenishment` to false after calling configure, since we own the options object and know that `GetTokenBucketLimiter` will do that anyways (but with an extra allocation).

Same with the other replenishing limiters.